src/Engine/Objects/Actor.cpp

```diff
@@ -4129,117 +4129,66 @@ void toggleActorGroupFlag(unsigned int uGroupID, ActorAttribute uFlag,
 
 //----- (004014E6) --------------------------------------------------------
 void Actor::MakeActorAIList_ODM() {
-    int v1;           // eax@4
-    unsigned int v7;  // ST20_4@10
-    int distance;     // edi@10
-                      //  int v10; // ebx@14
-    int v21;          // [sp+Ch] [bp-14h]@4
-    int v22;          // [sp+10h] [bp-10h]@4
+    std::vector<std::pair<int, int>> activeActorsDistances;
 
-    pParty->uFlags &= 0xFFFFFFCF;  // ~0x30
+    pParty->uFlags &= ~PARTY_FLAGS_1_ALERT_RED_OR_YELLOW;
 
-    ai_arrays_size = 0;
-    for (uint i = 0; i < pActors.size(); ++i) {
-        Actor *actor = &pActors[i];
-
-        actor->ResetFullAiState();  // ~0x400
-        if (!actor->CanAct()) {
-            actor->ResetActive();
+    for (int i = 0; i < pActors.size(); ++i) {
```

Comment:
Can replace with range-based for loop and use `Actor::id` where `i` is currently used.